ykrt/src/mt.rs

```diff
@@ -93,6 +93,10 @@ pub struct MT {
     /// The [Compiler] that will be used for compiling future `IRTrace`s. Note that this might not
     /// be the same as the compiler(s) used to compile past `IRTrace`s.
     compiler: Mutex<Arc<dyn Compiler>>,
+    /// A monotonically increasing integer that semi-uniquely identifies each compiled trace. This
+    /// is only useful for general debugging purposes, and must not be relied upon for semantic
+    /// correctness, because the IDs will wrap when the underlying `u64` overflows.
```

Comment:
Hah, I would assume you would have long run out of memory to compile further traces before you overflow this. :D

Reply:
Yes, this is paranoia writ large! I even briefly considered making it a `u128` before doing the maths. But, still, since we don't *need* to rely on it for semantic correctness, we might as well document that. That's one footgun our great-great-grandchildren won't encounter ;)

Reply:
You are very optimistic about the longevity of our software. :stuck_out_tongue: